src/registrar/models/domain.py

```diff
@@ -514,14 +1086,19 @@ def _fetch_cache(self, fetch_hosts=False, fetch_contacts=False):
                     cleaned["contacts"].append(
                         {k: v for k, v in contact.items() if v is not ...}
                     )
```

Comment:
Suggestion here is to get the public contact if it exists for the id, if not create that public contact for this domain. this is just to handle the possible case of things becoming out of sync between the registry and registrar. Feel free to discuss later.